core/user_api.php

```diff
@@ -997,7 +997,8 @@ function user_get_name( $p_user_id ) {
  * @return bool true to show, false otherwise.
  */
 function user_show_realname() {
-	return access_has_project_level( config_get( 'show_user_realname_threshold', null, null, ALL_PROJECTS ) );
+	return auth_is_user_authenticated() &&
```

Comment:
Can you explain please why the additional check for auth_is_user_authenticated() is needed?
I would expect that access_has_project_level() returns false if user is not logged in.

Reply:
That doesn't seem to be the case.  At the time when I implemented the fix I didn't want to change at lower level, though I considered it.